### PR TITLE
Scripts/Spells: fix Rupture, Eviscerate, Envenom, Slice and Dice

### PR DESCRIPTION
**Pull request by MaddyCrowle**
**Changes proposed:**

-  Rogue's Rupture, Eviscerate, Envenom, Slice and Dice are now take into account the combo points spent.

**Tests performed:**

Successfully built, tested in-game.

**Known issues and TODO list:** (add/remove lines as needed)

- Percs and procs related to these skills are not checked yet.

### CODE REVIEW

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 1]
```diff
@@ -36,46 +36,46 @@
 
 enum RogueSpells
 {
-    SPELL_ROGUE_ADRENALINE_RUSH                     = 13750,
```
Comment (Nyr97) [inline src/server/scripts/Spells/spell_rogue.cpp:39]: You messed up spacing.

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 2]
```diff
@@ -256,17 +256,16 @@ class spell_rog_envenom : public SpellScript
 {
     PrepareSpellScript(spell_rog_envenom);
 
-    void CalculateDamage(SpellEffIndex /*effIndex*/)
+    void CalculateDamage(SpellEffIndex effIndex)
     {
-        int32 damagePerCombo = GetHitDamage();
+        Optional<int32> cp = GetSpell()->GetPowerTypeCostAmount(POWER_COMBO_POINTS);
```
Comment (Nyr97) [inline src/server/scripts/Spells/spell_rogue.cpp:261]: comboPoints preferibly

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 3]
```diff
@@ -256,17 +256,16 @@ class spell_rog_envenom : public SpellScript
...
+        int32 ap = GetCaster()->GetTotalAttackPowerValue(BASE_ATTACK);
```
Comment (Nyr97) [inline src/server/scripts/Spells/spell_rogue.cpp:262]: attackPower preferably

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 4]
```diff
@@ -15,11 +15,11 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-/*
- * Scripts for spells with SPELLFAMILY_ROGUE and SPELLFAMILY_GENERIC spells used by rogue players.
- * Ordered alphabetically using scriptname.
- * Scriptnames of files in this file should be prefixed with "spell_rog_".
- */
+ /*
```
Comment (Aokromes) [inline src/server/scripts/Spells/spell_rogue.cpp:18]: revert this change.

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 5]
```diff
@@ -378,11 +376,11 @@ class spell_rog_killing_spree_aura : public AuraScript
     bool Validate(SpellInfo const* /*spellInfo*/) override
     {
         return ValidateSpellInfo(
-        {
-            SPELL_ROGUE_KILLING_SPREE_TELEPORT,
-            SPELL_ROGUE_KILLING_SPREE_WEAPON_DMG,
-            SPELL_ROGUE_KILLING_SPREE_DMG_BUFF
-        });
+            {
```
Comment (Aokromes) [inline src/server/scripts/Spells/spell_rogue.cpp:379]: revert this change

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 6]
```diff
@@ -773,15 +786,15 @@ class spell_rog_stealth : public AuraScript
...
         return ValidateSpellInfo(
-        {
-            SPELL_ROGUE_MASTER_OF_SUBTLETY_PASSIVE,
-            SPELL_ROGUE_MASTER_OF_SUBTLETY_DAMAGE_PERCENT,
-            SPELL_ROGUE_SANCTUARY,
-            SPELL_ROGUE_SHADOW_FOCUS,
-            SPELL_ROGUE_SHADOW_FOCUS_EFFECT,
-            SPELL_ROGUE_STEALTH_STEALTH_AURA,
-            SPELL_ROGUE_STEALTH_SHAPESHIFT_AURA
-        });
```
Comment (Aokromes) [inline src/server/scripts/Spells/spell_rogue.cpp:784]: revert this change.
